server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -135,7 +136,9 @@ private Map<String, UploadedSegmentMetadata> readLatestMetadataFile() throws IOE
 
     private Map<String, UploadedSegmentMetadata> readMetadataFile(String metadataFilename) throws IOException {
         try (IndexInput indexInput = remoteMetadataDirectory.openInput(metadataFilename, IOContext.DEFAULT)) {
-            RemoteSegmentMetadata metadata = metadataStreamWrapper.readStream(indexInput);
+            byte[] metadataBytes = new byte[(int) indexInput.length()];
+            indexInput.readBytes(metadataBytes, 0, (int) indexInput.length());
+            RemoteSegmentMetadata metadata = metadataStreamWrapper.readStream(new ByteArrayIndexInput(metadataFilename, metadataBytes));
```

Comment:
Why can't we provide `indexInput` as it is to `metadataStreamWrapper.readStream`?